### PR TITLE
travis-ci: Restructure build scripts and clean up

### PR DESCRIPTION
**Pull request by LNJ2**
This also adds caching for /var/cache/apt/archives, which will really speed up
the build, because then the slow launchpad servers aren't used anymore.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 dist: trusty
 install:
-  - ./utils/travis-dependencies.sh
+  - ./utils/travis/dependencies.sh
 script:
-  - cd utils
-  - source /opt/qt*/bin/qt*-env.sh
-  - ./travis-build.sh
+  - . utils/travis/common.sh
+  - env_setup
```
Comment (JBBgameich) [inline .travis.yml]: This will not work if the Qt 5.9 backports are not installed, like in the future for the click build

Comment (LNJ2) [reply .travis.yml]: The function checks if qt exsists!
https://github.com/KaidanIM/Kaidan/pull/208/files#diff-1c37644e56f5e80d16112e49505e1d1fR61

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+  - ./utils/travis/build.sh
```
Comment (JBBgameich) [inline .travis.yml]: Can you maybe also only run travis/build.sh if a cmake or qmake build is done? If we are already restrructuring everything ... 
I saw that you can run individual build commands for each matrix element in the minetest repository.

Comment (LNJ2) [reply .travis.yml]: Of course you can, and if you set one, it'll overwrite the default configuration.

File: .travis.yml (hunk 1 above)
Comment (JBBgameich) [reply .travis.yml]: Oh sorry, missed that :)

File: .travis.yml (hunk 2 above)
Comment (JBBgameich) [reply .travis.yml]: :+1:

File: utils/travis/common.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#!/bin/bash -e
+
+export CPUS_USED=2
+if [[ $(command -v nproc) != "" ]]; then
```
Comment (JBBgameich) [inline utils/travis/common.sh]: What about `if command -v nproc`; then?

Comment (LNJ2) [reply utils/travis/common.sh]: I think this outputted me the command path then, when I executed the script.

Comment (JBBgameich) [reply utils/travis/common.sh]: append `>/dev/null 2>&1`, so it's `if command -v nproc >/dev/null 2>&1; then`

Comment (LNJ2) [reply utils/travis/common.sh]: But that makes the code more complicated than necessary.

**Pull request merged**